### DISCUSSION TITLE
Update to current version

### DISCUSSION THREAD

**Discussion by nloshka** [firefly-iii/firefly-iii#6071] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

Debug information generated at 2022-05-13 10:07:55 Europe/Amsterdam for Firefly III version **5.5.11**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.5.11 |
| Firefly III API | 1.5.2 |
| PHP | 8.0.19 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | 28629bf3-d0fb-404b-a873-af978b1c355d |
| Using docker? | false |
| Telemetry | false |
| Layout |  |
| System TZ | Europe/Amsterdam |
| Browser TZ | Europe/Moscow |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, daily |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, sqlite |
| Current driver | mysql |
| DB version | 16 (exp. 16) |
| Login provider | eloquent  |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| Session start | 2022-05-01 00:00:00 |
| Session end | 2022-05-31 23:59:59 |
| Session first | 2018-12-08 00:00:00 |
| User ID | 1 |
| User language | ru_RU |
| User locale | ru_RU |
| Attempt at "ru_RU.utf8" | 'ru_RU.utf8' |
| Attempt at "ru_RU.UTF-8" | 'ru_RU.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36 |

### Solution

tell me how to properly upgrade from version 5.5.11 to the latest version?
I have just migrated all the content from one server to another, restored the work and updated PHP versions to PHP 8.0.19 (cli)
Can I upgrade from these versions or will I need to somehow install a new version and transfer the entered data through import export?
Thanks in advance for the detailed answer.

**Comment by JC5** [comment]
You should be able to upgrade easily

https://docs.firefly-iii.org/firefly-iii/installation/upgrade/

**Reply by nloshka** [reply]
git pull origin 5.7.6
rm -rf bootstrap/cache/*
rm -rf vendor/
composer install --no-scripts --no-dev
composer install --no-dev
php artisan migrate --seed
php artisan firefly-iii:decrypt-all
php artisan cache:clear
php artisan firefly-iii:upgrade-database
php artisan passport:install
php artisan cache:clear


--------------------
error 500

--------------------
sudo chown -R www-data:www-data firefly-iii
sudo chmod -R 775 firefly-iii/storage
 error 500

-----
:-(

**Reply by JC5** [reply]
I can't help you without knowing the exact error and the stack trace, thanks.

**Reply by nloshka** [reply]
what file do you need the information from?

**Reply by JC5** [reply]
https://docs.firefly-iii.org/firefly-iii/faq/other/#how-do-i-enable-debug-mode

**Comment by nloshka** [comment]
Target class [queue] does not exist.
Exceptions 2Stack Traces 2
Illuminate\Contracts\Container\
BindingResolutionException
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php ](https://buh.x512.cf/)(line 835)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php ](https://buh.x512.cf/)-> build (line 714)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php ](https://buh.x512.cf/)-> resolve (line 841)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php ](https://buh.x512.cf/)-> resolve (line 652)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php ](https://buh.x512.cf/)-> make (line 826)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php ](https://buh.x512.cf/)-> make (line 1365)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Bus/BusServiceProvider.php ](https://buh.x512.cf/)-> offsetGet (line 22)
BusServiceProvider->Illuminate\Bus\{closure}()
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php ](https://buh.x512.cf/)call_user_func (line 219)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php ](https://buh.x512.cf/)-> dispatchToQueue (line 77)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Bus/PendingDispatch.php ](https://buh.x512.cf/)-> dispatch (line 193)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php ](https://buh.x512.cf/)-> __destruct (line 389)
dispatch()
in [/var/www/buh.x512.cf/firefly-iii/app/Exceptions/Handler.php ](https://buh.x512.cf/)(line 171)
in [/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php ](https://buh.x512.cf/)-> report (line 88)
HandleExceptions->handleException()
ReflectionException
Class "queue" does not exist

**Reply by nloshka** [reply]
[2/2] BindingResolutionException
--
Illuminate\Contracts\Container\BindingResolutionException: Target class [queue] does not exist.    at /var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php:835   at Illuminate\Container\Container->build()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php:714)   at Illuminate\Container\Container->resolve()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:841)   at Illuminate\Foundation\Application->resolve()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php:652)   at Illuminate\Container\Container->make()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:826)   at Illuminate\Foundation\Application->make()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php:1365)   at Illuminate\Container\Container->offsetGet()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Bus/BusServiceProvider.php:22)   at Illuminate\Bus\BusServiceProvider->Illuminate\Bus\{closure}()   at call_user_func()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:219)   at Illuminate\Bus\Dispatcher->dispatchToQueue()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php:77)   at Illuminate\Bus\Dispatcher->dispatch()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Bus/PendingDispatch.php:193)   at Illuminate\Foundation\Bus\PendingDispatch->__destruct()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php:389)   at dispatch()      (/var/www/buh.x512.cf/firefly-iii/app/Exceptions/Handler.php:171)   at FireflyIII\Exceptions\Handler->report()      (/var/www/buh.x512.cf/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:88)   at Illuminate\Foundation\Bootstrap\HandleExceptions->handleException()

**Comment by nloshka** [comment]
[2022-05-20 14:48:36] local.ERROR: Exception is: {"class":"ErrorException","errorMessage":"require(\/var\/www\/buh.x512.cf\/firefly-iii\/vendor\/fideloper\/proxy\/src\/..\/config\/trustedproxy.php): Failed to open stream: No such file or directory","time":"Fri, 20 May 2022 14:48:36 +0200","file":"\/var\/www\/buh.x512.cf\/firefly-iii\/vendor\/laravel\/framework\/src\/Illuminate\/Support\/ServiceProvider.php","line":130,"code":0,"version":"5.5.11","url":"https:\/\/buh.x512.cf","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/99.0.4844.74 Safari\/537.36","json":true}  
[2022-05-20 14:48:36] local.ERROR: Exception is: {"class":"Error","errorMessage":"Class \"Laravel\\SerializableClosure\\SerializableClosure\" not found","time":"Fri, 20 May 2022 14:48:36 +0200","file":"\/var\/www\/buh.x512.cf\/firefly-iii\/vendor\/laravel\/framework\/src\/Illuminate\/Queue\/QueueServiceProvider.php","line":49,"code":0,"version":"5.5.11","url":"https:\/\/buh.x512.cf","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/99.0.4844.74 Safari\/537.36","json":true}

**Comment by nloshka** [comment]
# php artisan migrate --seed
Migrating: 2021_07_05_193044_drop_tele_table
Migrated:  2021_07_05_193044_drop_tele_table (19.24ms)
Migrating: 2021_08_28_073733_user_groups

   Illuminate\Database\QueryException 

  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'user_groups' already exists (SQL: create table `user_groups` (`id` bigint unsigned not null auto_increment primary key, `created_at` timestamp null, `updated_at` timestamp null, `deleted_at` timestamp null, `title` varchar(255) not null) default character set utf8mb4 collate 'utf8mb4_unicode_ci' engine = InnoDB)

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:742
    738▕         // If an exception occurs when attempting to run a query, we'll format the error
    739▕         // message to include the bindings with SQL, which will make this exception a
    740▕         // lot more helpful to the developer instead of just the database's errors.
    741▕         catch (Exception $e) {
  ➜ 742▕             throw new QueryException(
    743▕                 $query, $this->prepareBindings($bindings), $e
    744▕             );
    745▕         }
    746▕     }

      +9 vendor frames 
  10  database/migrations/2021_08_28_073733_user_groups.php:94
      Illuminate\Support\Facades\Facade::__callStatic()

      +22 vendor frames 
  33  artisan:37
      Illuminate\Foundation\Console\Kernel::handle()

**Comment by nloshka** [comment]
It might be easier to do a data export,
then reinstall system as new
and import data into the new system?
Is it possible to transfer everything without loss of data?

**Reply by JC5** [reply]
Sorry, I can't make heads or tails of these logs. It seems to point to badly installed packages.

You can get a new installation and then link it to the old database. That should upgrade it automatically.

**Reply by nloshka** [reply]
Do you mean that there are not enough packages for php? Everything that is described in the requirements, I installed the necessary PHP packages on the server. Can you find out exactly what is missing? What packages are missing?

**Comment by nloshka** [comment]
I was able to upgrade using a fresh install.
Maybe someone will come in handy.
update process:
Performing a new installation
Migrating the database from the old database to the new one
Delete the entire table group_memberships user_groups user_roles
Run: php artisan firefly-iii:upgrade-database
And the migration is done.
Thanks, maybe some experience will help.

**Reply by JC5** [reply]
Thanks for letting me know this worked!
